nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/AbstractCredentialsStrategy.java

```diff
@@ -80,22 +81,27 @@ public Collection<ValidationResult> validate(final ValidationContext validationC
         return validationFailureResults;
     }
 
-    public abstract AWSCredentialsProvider getCredentialsProvider(Map<PropertyDescriptor, String> properties);
+    public abstract AWSCredentialsProvider getCredentialsProvider(final Map<PropertyDescriptor, String> properties);
 
     public String getName() {
         return name;
     }
 
 
     @Override
-    public boolean canCreateDerivedCredential(Map<PropertyDescriptor, String> properties) {
+    public boolean canCreateDerivedCredential(final Map<PropertyDescriptor, String> properties) {
         return false;
     }
 
     @Override
-    public AWSCredentialsProvider getDerivedCredentialsProvider(Map<PropertyDescriptor, String> properties,
-                                                                AWSCredentialsProvider primaryCredentialsProvider) {
+    public AWSCredentialsProvider getDerivedCredentialsProvider(final Map<PropertyDescriptor, String> properties,
+                                                                final AWSCredentialsProvider primaryCredentialsProvider) {
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public AwsCredentialsProvider getDerivedAwsCredentialsProvider(final Map<PropertyDescriptor, String> properties,
+                                                                   AwsCredentialsProvider primaryCredentialsProvider) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Although this follows the above pattern, is this necessary? Seems like it is best left unimplemented since the class is abstract.